src/App.tsx

```diff
@@ -15,33 +14,110 @@ const goodsFromServer: string[] = [
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: React.FC = () => {
+  const [isStarted, setIsStarted] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+  );
+
+  return (
+    <div className="App">
+      {!isStarted && (
+        <button
+          type="button"
+          className="button is-primary button-start"
+          onClick={() => setIsStarted(true)}
+        >
+          Start
+        </button>
+      )}
+
+      {isStarted && (
+        <div className="box">
+          <ul className="Goods">
+            {visibleGoods.map(good => (
+              <li className="Goods__item" key={good}>
+                {good}
+              </li>
+            ))}
+          </ul>
+
+          <div className="buttons">
+            <button
+              type="button"
+              className="button is-primary"
+              onClick={() => setSortType(SortType.ALPHABET)}
+            >
+              Sort alphabetically
+            </button>
+
+            <button
+              type="button"
+              className="button is-primary"
+              onClick={() => setSortType(SortType.LENGTH)}
+            >
+              Sort by length
+            </button>
+
+            <button
+              type="button"
+              className="button is-primary"
+              onClick={() => setIsReversed(!isReversed)}
+            >
+              Reverse
+            </button>
+
+            <button
+              type="button"
+              className="button is-primary"
+              onClick={() => {
+                setSortType(SortType.NONE);
+                setIsReversed(false);
+              }}
+            >
+              Reset
```

Comment:
For me it's better too make a new function reset() for onClick, it will look more clear.